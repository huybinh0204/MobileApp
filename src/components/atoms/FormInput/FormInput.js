import { ErrorMessage } from '@hookform/error-message';
import PropTypes from 'prop-types';
import React from 'react';
import { useController, useFormContext } from 'react-hook-form';
import Input from '_components/atoms/Input/Input';
import { Container, ErrorText } from './FormInput.styles';

const FormInput = ({ name, variant, ...rest }) => {
  const { control, formState } = useFormContext();
  const { field } = useController({ name, control });
  const error = formState.errors?.[name];

  return (
    <Container>
      <Input error={error} onChangeText={field.onChange} value={field.value} {...rest} />
      <ErrorMessage as={<ErrorText />} errors={formState.errors} name={name} />
    </Container>
  );
};

FormInput.propTypes = {
  name: PropTypes.string.isRequired,
};

export default FormInput;
