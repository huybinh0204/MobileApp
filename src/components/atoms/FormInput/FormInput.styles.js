import styled from '@emotion/native';
import BaseText from '_components/atoms/BaseText/BaseText';

export const Container = styled.View(({ theme }) => ({
  marginBottom: theme.spacing.l,
}));

export const ErrorText = styled(BaseText)(({ theme }) => ({
  ...theme.typography.tinyContentRegular,
  color: theme.colors.epsilon500,
  marginVertical: theme.spacing.xs,
}));
