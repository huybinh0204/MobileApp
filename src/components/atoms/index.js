export { default as AccountNumber } from './AccountNumber/AccountNumber';
export { default as Badge } from './Badge/Badge';
export { default as Balance } from './Balance/Balance';
export { default as BaseFormInput } from './BaseFormInput/BaseFormInput';
export { default as BaseText } from './BaseText/BaseText';
export { default as BottomSheet } from './BottomSheet/BottomSheet';
export { default as Card } from './Card/Card';
export { default as CurrencyInput } from './CurrencyInput/CurrencyInput';
export { default as Dash } from './Dash/Dash';
export { default as Form } from './Form/Form';
export { default as FormInput } from './FormInput/FormInput';
export { default as FormSubmitButton } from './FormSubmitButton/FormSubmitButton';
export { default as GeneralContainer } from './GeneralContainer/GeneralContainer';
export { default as IconSvg } from './IconSvg/IconSvg';
export { default as Link } from './Link/Link';
export { default as MainButton } from './MainButton/MainButton';
export { default as Money } from './Money/Money';
export { default as PdfView } from './PdfView/PdfView';
export { default as Plaid } from './Plaid/Plaid';
export { default as PressableListItem } from './PressableListItem/PressableListItem';
export { default as ProgressBar } from './ProgressBar/ProgressBar';
export { default as SegmentedTextInput } from './SegmentedTextInput/SegmentedTextInput';
export { default as SelectButton } from './SelectButton/SelectButton';
export { default as Switch } from './Switch/Switch';
export { default as TabItem } from './TabItem/TabItem';
export { default as Toast } from './Toast/Toast';
export { default as TransactionItem } from './TransactionItem/TransactionItem';
