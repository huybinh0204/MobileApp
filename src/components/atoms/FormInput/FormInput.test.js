import { render } from '@testing-library/react-native';
import React from 'react';
import { useForm } from 'react-hook-form';
import { Form } from '_components/atoms';
import Theme from '_utilities/Theme';
import FormInput from './FormInput';

const RenderFormInput = (innerProps) => {
  const formMethods = useForm({ defaultValues: { test: '' } });

  return (
    <Theme>
      <Form formMethods={formMethods}>
        <FormInput {...innerProps} />
      </Form>
    </Theme>
  );
};

describe('FromInput', () => {
  it('should match the snapshot', () => {
    const { toJSON } = render(<RenderFormInput name="test" placeholder="test" />);
    expect(toJSON()).toMatchSnapshot();
  });
});
