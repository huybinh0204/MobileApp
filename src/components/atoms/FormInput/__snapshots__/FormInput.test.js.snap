// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FromInput should match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 87,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 6,
          "borderWidth": 2,
          "flexDirection": "row",
          "width": "100%",
          "zIndex": 0,
        },
        Object {
          "borderColor": undefined,
        },
      ]
    }
  >
    <BVLinearGradient
      colors={
        Array [
          309817028,
          318208316,
        ]
      }
      endPoint={
        Object {
          "x": 0.75,
          "y": 0.5,
        }
      }
      locations={null}
      startPoint={
        Object {
          "x": 0.25,
          "y": 0.5,
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 6,
            "height": "100%",
            "position": "absolute",
            "width": "100%",
            "zIndex": 1,
          },
          Object {
            "opacity": undefined,
          },
        ]
      }
    />
    <Text
      allowFontScaling={false}
      onLayout={[Function]}
      style={
        Array [
          Object {
            "bottom": 0,
            "fontFamily": "Poppins",
            "fontSize": 29,
            "left": 25.5,
            "lineHeight": 40,
            "position": "absolute",
            "zIndex": 2,
          },
          Object {
            "color": undefined,
            "transform": Array [
              Object {
                "translateY": undefined,
              },
              Object {
                "translateX": undefined,
              },
              Object {
                "scale": undefined,
              },
            ],
          },
        ]
      }
    >
      test
    </Text>
    <TextInput
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      style={
        Array [
          Object {
            "color": "#1F1D25",
            "flex": 1,
            "fontFamily": "Poppins-SemiBold",
            "fontSize": 29,
            "lineHeight": 40,
            "paddingBottom": 18,
            "paddingHorizontal": 25.5,
            "paddingTop": 43.5,
            "zIndex": 3,
          },
          undefined,
        ]
      }
      value=""
    />
  </View>
</View>
`;
